### ISSUE TITLE
[Metricbeat] Reduce "compose.EnsureUp" timeout in tests to 60 seconds

### ISSUE THREAD

**Issue by sayden** [elastic/beats#10876]
I have seen that in many tests we have a timeout of 5 minutes waiting for a compose service to be ready like here https://github.com/elastic/beats/blob/master/metricbeat/module/jolokia/jmx/jmx_integration_test.go#L49

I think we can safely reduce this time to 1 minute or even 30 seconds. The reason behind this is that I see often flaky tests that are failing after those 5 minutes passed (and probably it would be the same with 1 minute or 1 hour because it's a problem with the Docker setup of the container itself).

@ruflin @jsoriano @kaiyan-sheng @exekias WDYT?

**Comment by sayden** [comment]
Example, fail after 302.932 seconds:

```
12:06:58 ok  	github.com/elastic/beats/metricbeat/module/zookeeper/mntr	10.392s	coverage: 84.0% of statements
12:10:37 command [go test -race -tags integration -cover -coverprofile /tmp/gotestcover-421953353 github.com/elastic/beats/metricbeat/module/rabbitmq/node]: exit status 1
12:10:37 Building rabbitmq
12:10:37 Step 1/4 : FROM rabbitmq:3.7.4-management
12:10:37  ---> 7d286259089f
12:10:37 Step 2/4 : RUN apt-get update && apt-get install -y netcat && apt-get clean
12:10:37  ---> Using cache
12:10:37  ---> 698f251a9588
12:10:37 Step 3/4 : HEALTHCHECK --interval=1s --retries=90 CMD nc -w 1 -v 127.0.0.1 15672 </dev/null
12:10:37  ---> Using cache
12:10:37  ---> 654021330b21
12:10:37 Step 4/4 : EXPOSE 15672
12:10:37  ---> Using cache
12:10:37  ---> 95affc5052bc
12:10:37 Successfully built 95affc5052bc
12:10:37 Successfully tagged metricbeat97f33c937432dee1235f6e4e23541a95e5cab575_rabbitmq:latest
12:10:37 Creating metricbeat97f33c937432dee1235f6e4e23541a95e5cab575_rabbitmq_1 ... 
12:10:37 
12:10:37 Creating metricbeat97f33c937432dee1235f6e4e23541a95e5cab575_rabbitmq_1 ... done
12:10:37 --- FAIL: TestData (302.74s)
12:10:37 	compose.go:64: Timeout waiting for services to be healthy
12:10:37 FAIL
12:10:37 coverage: 65.4% of statements
12:10:37 FAIL	github.com/elastic/beats/metricbeat/module/rabbitmq/node	302.932s
12:10:37 test failed
12:10:37 make: *** [integration-tests] Error 1
12:10:37 ../libbeat/scripts/Makefile:198: recipe for target 'integration-tests' failed
12:10:37 make[1]: *** [integration-tests-environment] Error 2
12:10:37 make: *** [testsuite] Error 2
12:10:37 make[1]: Leaving directory `/var/lib/jenkins/workspace/elastic+beats+pull-request+multijob-linux/beat/metricbeat/label/linux-immutable/src/github.com/elastic/beats/metricbeat'
```

**Comment by jsoriano** [comment]
@sayden I am totally +1 to reduce this kind of times. But we should do it carefuly, I think there are services that can take several minutes to be ready, but they are probably few.

Maybe you could open a PR with the change and do some CI runs to see what services fail to start with the new timeout. We could use `EnsureUpWithTimeout()` to give a bigger timeout to them.

**Comment by sayden** [comment]
Yeap, my main concern is that we could reduce the CI time but increase the number of flaky tests because there are actually few services that, as you say, might take quite a while to setup.

But, your idea of playing a bit with a PR is good. There are some hours during the day that Jenkins is a bit more "idle" where I can trigger few builds.

**Comment by sayden** [comment]
First attempt. Exactly the same test, module and metricset (TestData in RabbitMQ/Node) failed after 1 minute instead of 5 minutes.

```
14:05:25 command [go test -race -tags integration -cover -coverprofile /tmp/gotestcover-206026902 github.com/elastic/beats/metricbeat/module/rabbitmq/node]: exit status 1
14:05:25 Building rabbitmq
14:05:25 Step 1/4 : FROM rabbitmq:3.7.4-management
14:05:25  ---> 7d286259089f
14:05:25 Step 2/4 : RUN apt-get update && apt-get install -y netcat && apt-get clean
14:05:25  ---> Using cache
14:05:25  ---> e042a0871e23
14:05:25 Step 3/4 : HEALTHCHECK --interval=1s --retries=90 CMD nc -w 1 -v 127.0.0.1 15672 </dev/null
14:05:25  ---> Using cache
14:05:25  ---> 7fe1201decd5
14:05:25 Step 4/4 : EXPOSE 15672
14:05:25  ---> Using cache
14:05:25  ---> ea7a48600d4b
14:05:25 
14:05:25 Successfully built ea7a48600d4b
14:05:25 Successfully tagged metricbeataca1092a979c12b044e48cca3b3b58c672574580_rabbitmq:latest
14:05:25 Creating metricbeataca1092a979c12b044e48cca3b3b58c672574580_rabbitmq_1 ... 
14:05:25 
14:05:25 Creating metricbeataca1092a979c12b044e48cca3b3b58c672574580_rabbitmq_1 ... done
14:05:25 --- FAIL: TestData (64.12s)
14:05:25 	compose.go:64: Timeout waiting for services to be healthy
14:05:25 FAIL
14:05:25 coverage: 65.4% of statements
14:05:25 FAIL	github.com/elastic/beats/metricbeat/module/rabbitmq/node	64.305s
```

**Comment by ruflin** [comment]
+1 on reducing the default but make it possible to overwrite (I think this is already the case). Especially for example Kibana sometimes needs more then 5mins on slow machines.

**Comment by sayden** [comment]
"Fixed" and merged in #10894 
The default timeout has been reduced to 60 seconds and it's overriden to 570 seconds in Kibana and Logstash
